### ISSUE TITLE
Update user-friendly name for specimen_from_organism schema

### ISSUE THREAD

**Issue by malloryfreeberg** [HumanCellAtlas/metadata-schema#991]
**For which schema is a change/update being suggested?**

I would like to request an update to the `specimen_from_organism.json` schema.

**What should the change/update be?**

I would like to change the user-friendly name of the specimen from organism schema to "Primary tissue" to more accurately describe that the specimen schema is to be used to describe a primary tissue sample taken directly from a donor/mouse.

Related to https://github.com/HumanCellAtlas/metadata-schema/issues/389 which proposed to change the whole schema name (a major and disruptive change). This ticket is just for updating the user-friendly name of the schema.

**What new field(s) need to be changed/added?**

No new fields.

**Why is the change requested?**

I would like the Data Browser to be able to display the "type" of sample in a project as either "Organoid", "Cell line", or "Primary tissue", which are the 3 major categories of sample types we have. The terms "Organoid" and "Cell line" can be retrieved from the user-friendly term of the corresponding schemas, but for "Primary tissue", the schema name is actually "Specimen from organism", which is a bit confusing. Changing the user-friendly name of the specimen schema is a patch update that will provide a mechanism for the Browser to retrieve the standardized sample type names.

**Comment by malloryfreeberg** [comment]
Thoughts @hannes-ucsc @noopdog ?

The idea is to go from this:

<img width="561" alt="Screen Shot 2019-05-17 at 12 35 08" src="https://user-images.githubusercontent.com/10993501/57925542-797f5800-78a0-11e9-849c-14debf1d717f.png">

to more like this:

<img width="561" alt="Screen Shot 2019-05-17 at 12 35 08 copy" src="https://user-images.githubusercontent.com/10993501/57925547-80a66600-78a0-11e9-9926-28c3ebca2111.png">

@hannes-ucsc I know you had some thoughts on what to replace "specimen" with (biospecimen, tissue, tissue sample). We can change the actual name of the schema later (major change) but change the user-friendly name now to help with displays.

**Comment by hannes-ucsc** [comment]
Good idea. See https://github.com/DataBiosphere/azul/issues/1026

**Comment by hannes-ucsc** [comment]
I have to back track from my initial enthusiasm. The column value in the data browser isn't the only place where our users encounter the `specimen_from_organism` misnomer. It manifests itself in the manifest and the full metadata TSV. I worry that this would actually increase the confusion among our users. We introduced the concept of "user friendly" name but does that mean that our schema uses "user-unfriendly" names? We do expect users to consume the metadata, do we not? Above all, I (and our users) value low entropy and consistency. I think this change would increase entropy and reduce consistency.

Furthermore, while I look forward to the renaming of the actual metadata entity—and I think it could be absorbed relatively easily—I don't want to spend any time on a cosmetic issue that I'll have to redo when the actual entity gets renamed.

One could still argue that this could be implemented at the presentation layer in the Data Browser. Not for me to decide. The entropy and consistency concern however would still apply.

**Comment by diekhans** [comment]
`display_name` would have been a better name of the field than `user_friendly`.  The current schema entity structure was developed rather quickly and does need review, especially from a consumer point of view.  It will be improved, but not in a `piecemeal` approach.

However, until we get a system in place to migrate entities to new metadata schema versions,  these changes become quite painful, because both old and new entities need to be supported.   Updating the `used_friendly` name is a non-breaking change, so it gives us a chance to improve usability quickly.

For `user_unfriendly`, we have UUIDs ;-)

**Comment by hannes-ucsc** [comment]
+1 on `display_name` but having `display_name` be "Primary Tissue" while the entity is called `specimen_from_organism` too far a leap, IMO.

**Comment by diekhans** [comment]
there is already a ticket on specimen_from_organism being a bad name:
https://github.com/HumanCellAtlas/metadata-schema/issues/389

I don't think you will get any argument about that.  We don't have the tools to make it not painful yet .

**Comment by diekhans** [comment]
I created a meta-ticket to collect these kinds of issues for review:

[Do a top-down review of metadata entity names as structure. #994](https://github.com/HumanCellAtlas/metadata-schema/issues/994)

this is already in there, but feel free to complain about similar issues

**Comment by hannes-ucsc** [comment]
@malloryfreeberg I think I got a little confused. This is a presentational issue that Azul should not be involved in. Azul will convert from "specimen" to "primary_tissue" when the actual entity is being renamed. That way Azul's service reponse will be consistent with the manifest and metadata TSVs that it generates. I would caution against doing a cosmetic renaming at the presentation layer (i.e. Data Browser but that's not for me to decide).

**Comment by diekhans** [comment]
@hannes-ucsc, I am confused.  where are you getting the name "specimen" from?  We should always be looking up display names, not translating programming names.  We might need to add something for you look up.

**Comment by malloryfreeberg** [comment]
@diekhans @hannes-ucsc Azul/Browser should be getting the display name from the schema using the `title` attribute. E.g.:

```
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "description": "Information about the cell line or cell culture biomaterial.",
    "additionalProperties": false,
    "title": "Cell line",
    "name": "cell_line",
```

I don't think Azul/Browser are doing this at moment, because instead of "Cell line" I see "cellLine" in the Browser. User-unfriendly names are the programmatic names (e.g. "cell_line").

Azul/Browser should be able to get all display names by introspecting the schema and not have to hard-code names (like "cellLine").

**Comment by hannes-ucsc** [comment]
@diekhans, Azul does not use display names because it is not a GUI. It's a service designed for programmatic consumption and should therefore be consistent with the canonical naming of entities and properties as defined in the schema. We may provide an endpoint that the Data Browser can use to determine the display name mapping for all fields. Because Azul indexes all metadata, it is in a good position to populate an exhaustive mapping across all schema versions found in the wild.

The use of `specimen` has historical reasons. We inherited it from our predecessors and never  changed it to `specimen_from_organism` because it seemed awkwardly long and a discussion  of renaming it had already been started.

**Comment by hannes-ucsc** [comment]
@malloryfreeberg, agreed. In this case would be helpful, though, to distinguish between Azul and the Data Browser. I think Azul should not use the `title` property from the schema other than to back a service endpoint that enables the Data Browser to determine the title of any given entity or property. For example, for a sample that is a cell line, the Azul service response should read `…, {"sample": [{ …, "type": "cell_line"}, …], "cell_line": [{…, "model_organ": …, …}, …], …}'. The data browser should translate `cell_line` to `"Cell line"` before displaying that response. The translation will be driven by a mapping that was retrieved from Azul before any other requests are made to it.

**Comment by diekhans** [comment]
@hannes-ucsc  I though azul was the data browser

what do I know???

yes, if it is a programmatic api it should use the field names as defined.  Until we have schema migration, we need to go slow (just like everything cross box) 👎

**Comment by zperova** [comment]
Needs review @lauraclarke

**Comment by lauraclarke** [comment]
This has come back up here https://github.com/HumanCellAtlas/data-browser/issues/863#issuecomment-548838232

It would be good to move this discussion forward so we can plan any changes that are needed alongside changes which would be needed to introduce the library entity

**Comment by diekhans** [comment]
the demod work address user_friendly names that are defined in modules

**Comment by ESapenaVentura** [comment]
Closing the ticket as the discussion didn't move forward @clairerye

**Comment by hannes-ucsc** [comment]
A stalled discussion isn't a good enough reason for closing a ticket. It's a reason for picking up the discussion.

As @lauraclarke [pointed out](https://github.com/HumanCellAtlas/metadata-schema/issues/991#issuecomment-549161363), users [still take issue with this](https://github.com/HumanCellAtlas/data-browser/issues/863). The data browser shouldn't rename the sample type without the corresponding schema types being named consistently.

Changing the names for the sample biomaterials schema types ("specimen" to "primary tissue", for example) would be a fairly invasive change that may not be worth the gain in clarity on the consumer side. I would understand if we decided not to address this, but I think we should be honest and state that explicitly as the reason for closing.

**Comment by ESapenaVentura** [comment]
The reasoning behind closing the ticket is to clear the backlog, this ticket has not had any updates (And I have not heard about it being addressed) in almost 2 years.

I am very happy to re-take the discussion and keep the ticket open if deemed necessary. I have to admit I didn't have time to look very thoroughly through each and every ticket.

I'll add the `task` label and will try to bring it into attention to the team!
